### PR TITLE
Further adapt integration tests to vert.x WebClient API

### PR DESCRIPTION
**Pull request by sophokles73**
The standard features of WebClient are now being used for specifying
predicates on the HTTP response. All operations of the CrudHttpClient
now return a Future<HttpResponse<Buffer>> so that all clients can get
access to the full response instead of just the headers or buffer
respectively.

### CODE REVIEW

File: tests/src/test/java/org/eclipse/hono/tests/DeviceRegistryHttpClient.java [hunk 1]
```diff
@@ -272,14 +277,13 @@ private static String registrationWithoutIdUri(final String tenant) {
      *
      * @param tenantId The tenant to get information for.
      * @param expectedStatusCode The status code indicating a successful outcome.
-     * @return A future indicating the outcome of the operation. The future will contain the response payload if the
-     *         response contained the expected status code. Otherwise the future will fail with a
-     *         {@link org.eclipse.hono.client.ServiceInvocationException}.
```
Comment (calohmn) [inline tests/src/test/java/org/eclipse/hono/tests/DeviceRegistryHttpClient.java:277]: The removal of the `with a {@link org.eclipse.hono.client.ServiceInvocationException}` part is not consistent in this class yet - it's still in there for several methods.

File: tests/src/test/java/org/eclipse/hono/tests/CrudHttpClient.java [hunk 2]
```diff
@@ -171,19 +166,24 @@ public CrudHttpClient(final Vertx vertx, final HttpClientOptions options) {
...
-     * @param checkCorsHeaders Whether to set and check CORS headers.
-     * @return A future that will succeed if the predicate evaluates to {@code true}.
-     * @throws NullPointerException if URI, content type or predicate are {@code null}.
+     * @param successPredicates Checks on the HTTP response that need to pass for the request
+     *                          to be considered successful.
+     * @return A future indicating the outcome of the request. The future will be completed with the
+     *         HTTP response if all checks on the response have succeeded.
+     *         Otherwise the future will be failed with the error produced by the first failing
+     *         predicate.
+     * @throws NullPointerException if URI or content type are {@code null}.
```
Comment (calohmn) [inline tests/src/test/java/org/eclipse/hono/tests/CrudHttpClient.java:175]: No NPE actually thrown if contentType is `null`.

**Comment by sophokles73** [comment]
@calohmn  I have fixed the JavaDoc ...

**Review by calohmn — APPROVED** [review]
LGTM

**Pull request merged**